hudi-utilities/src/main/java/org/apache/hudi/utilities/transform/ChainedTransformer.java

```diff
@@ -46,9 +53,33 @@ public List<String> getTransformersNames() {
   @Override
   public Dataset<Row> apply(JavaSparkContext jsc, SparkSession sparkSession, Dataset<Row> rowDataset, TypedProperties properties) {
     Dataset<Row> dataset = rowDataset;
+    boolean isErrorTableEnabled = properties.getBoolean(ERROR_TABLE_ENABLED.key(), ERROR_TABLE_ENABLED.defaultValue());
+    if (isErrorTableEnabled && !isErrorRecordPresent(dataset)) {
+      dataset = dataset.withColumn(ERROR_TABLE_CURRUPT_RECORD_COL_NAME, lit(null));
+    }
     for (Transformer t : transformers) {
       dataset = t.apply(jsc, sparkSession, dataset, properties);
+      // validate in every stage to ensure it's not dropped by one of the transformer and added by next transformer.
+      validate(dataset, isErrorTableEnabled);
```

Comment:
if using error table requires some column to be present, we should enforce this rule within error table itself. Transformer should not need to know about whether error table enabled or not. There could be a cleaner solution to solve this.

Reply:
changed implementation to use separate chainedTransformer for error table .